### DISCUSSION TITLE
Issue converting scanpy h5ad file to h5seurat in R

### DISCUSSION THREAD

**Discussion by LJ-dotcom** [satijalab/seurat#9430] [Q&A] [answered]
Hi, I'm trying to use R to work with snRNAseq data in the GTEx portal found at this link: https://www.gtexportal.org/home/downloads/adult-gtex/single_cell
The file is called GTEx_8_tissues_snRNAseq_atlas_071421.public_obs.h5ad

When I run this code in R:

library(Seurat)
library(SeuratData)
library(SeuratDisk)
Convert("~/Desktop/GTEx_8_tissues_snRNAseq_atlas_071421.public_obs.h5ad", dest = "h5seurat", overwrite = TRUE)
sc <- LoadH5Seurat("~/Desktop/GTEx_8_tissues_snRNAseq_atlas_071421.public_obs.h5seurat")

I get the following errors:

Warning: Unknown file type: h5ad
Warning: 'assay' not set, setting to 'RNA'
Creating h5Seurat file for version 3.1.5.9900
Adding X as data
Adding raw/X as counts
Adding meta.features from raw/var
Error in dfile$obj_copy_from(src_loc = source, src_name = "obs", dst_name = "meta.data") : 
  HDF5-API Errors:
    error #000: ../../src/hdf5-1.12.2/src/H5Ocopy.c in H5Ocopy(): line 240: unable to copy object
        class: HDF5
        major: Object header
        minor: Unable to copy object

    error #001: ../../src/hdf5-1.12.2/src/H5VLcallback.c in H5VL_object_copy(): line 5495: object copy failed
        class: HDF5
        major: Virtual Object Layer
        minor: Unable to copy object

    error #002: ../../src/hdf5-1.12.2/src/H5VLcallback.c in H5VL__object_copy(): line 5456: object copy failed
        class: HDF5
        major: Virtual Object Layer
        minor: Unable to copy object

    error #003: ../../src/hdf5-1.12.2/src/H5VLnative_object.c in H5VL__native_object_copy(): line 125: unable to copy object
        class: HDF5
        major: Object header
        minor: Unable to copy object

    error #004: ../../src/hdf5-1.12.2/src/H5Ocopy.c in H5O__copy(): line 301: unable to copy object
        class: HDF5
        major: Object header
        minor: Unab
Error in private$closeFun(id) : HDF5-API Errors:
    error #000: ../../src/hdf5-1.12.2/src/H5F.c in H5Fclose(): line 711: decrementing file ID failed
        class: HDF5
        major: File accessibility
        minor: Unable to close file

    error #001: ../../src/hdf5-1.12.2/src/H5Iint.c in H5I_dec_app_ref(): line 1018: can't decrement ID ref count
        class: HDF5
        major: Object atom
        minor: Unable to decrement reference count

    error #002: ../../src/hdf5-1.12.2/src/H5Fint.c in H5F__close_cb(): line 251: unable to close file
        class: HDF5
        major: File accessibility
        minor: Unable to close file

    error #003: ../../src/hdf5-1.12.2/src/H5VLcallback.c in H5VL_file_close(): line 3983: file close failed
        class: HDF5
        major: Virtual Object Layer
        minor: Unable to close file

    error #004: ../../src/hdf5-1.12.2/src/H5VLcallback.c in H5VL__file_close(): line 3952: file close failed
        class: HDF5
        major: Virtual Object Layer
        min

A h5seurat file is produced despite the errors, and when I ran this:
library(rhdf5)
h5ls("~/Desktop/GTEx_8_tissues_snRNAseq_atlas_071421.public_obs.h5seurat")

I got a table with the following column names: group, name, otype, dclass, and dim. The table has 57 rows. 

I haven't worked with this file format before, so I am unsure if this is what I am supposed to see or if the errors are causing issues.

Has anyone worked with this specific file before, or does anyone have an idea as to why these errors are occurring? 
Thank you so much for taking the time to read through this!

**Comment by denvercal1234GitHub** [comment]
Check out https://github.com/denvercal1234GitHub/anndata_Seurat_Utilities to see if it helps your case.

**Reply by samuel-marsh** [reply]
As SeuratDisk is no longer maintained or supported it is probably best not to use in conversion/loading of objects especially given large changes to Seurat object structure since its end of maintainence.

Best,
Sam

**Comment by samuel-marsh** [comment, marked as answer]
Th scverse team has released an official R package to perform such conversions.
https://github.com/scverse/anndataR
